### ISSUE TITLE
Farm mgmt: Automation for farm network creation

### ISSUE THREAD

**Issue by andhartl** [threefoldfoundation/tf_farmer_management_tools#1]
Use ansible to create automated farm network setup. automation will only work for one predefined switch. 
Jan to confirm
- [x] type of switch (cumulus, fs.com)
- [x] network design

Might also need to create SA

**Comment by m-motawea** [comment]
blocked on switch platform

**Comment by weynandkuijpers** [comment]
@m-motawea You mean the selection of the switch platform? Does Jan need to do this?

**Comment by m-motawea** [comment]
@weynandkuijpers yes

**Comment by m-motawea** [comment]
```
all fs.com:   s8050 and s5850 for backend
s3800 for mgmt, +1 s3900 for aggregating the s3800
```

unfortunately, I have no way to simulate these switches locally. maybe we can have a jumpserver connected to an old unused switch or something?

**Comment by weynandkuijpers** [comment]
reached out to Jan for this - somehow he is not part of the foundation organisation.  Send invite.

**Comment by andhartl** [comment]
we use fs.com switches in the Lochristi datacenter. It should be possible for Jan to give @m-motawea access.

**Comment by sasha-astiadi** [comment]
@m-motawea is it still blocked? did you gain access to switches

**Comment by weynandkuijpers** [comment]
@delandtj and @m-motawea Any update / progress?

**Comment by m-motawea** [comment]
for now we have basic roles for LACP, VLANs, IP MAnagment, Trunking and ports state in both FSOS and Cumulus.
we also have DHPC server config roles for cumulus (no dhcp server available on the switch s3900 that we currently access)

sample playbooks at:

- Cumulus: https://github.com/threefoldfoundation/tf_farmer_management_tools/blob/development/docs/roles/cumulus/playbook.yaml

- FSOS: https://github.com/threefoldfoundation/tf_farmer_management_tools/blob/development/docs/roles/fsos/playbook.yaml

**Comment by weynandkuijpers** [comment]
Okay - how much more is needed in order to finish this issue?

**Comment by m-motawea** [comment]
@weynandkuijpers we need the environment (and the design) that we'll use for demo (lab) https://github.com/threefoldfoundation/tf_farmer_management_tools/issues/4.  I think we covered most of the required roles. but we need to do the actual demo which requires a full environment to test against.

**Comment by weynandkuijpers** [comment]
sounds good - can you (screen) record the demo?

**Comment by weynandkuijpers** [comment]
Can we put this in verification - and verification being the demo done?

**Comment by andhartl** [comment]
still missing network design @m-motawea , right?

**Comment by m-motawea** [comment]
@delandtj  need the remaining 8 mac addresses for the servers to be used in validation by @hossnys

**Comment by andhartl** [comment]
talked to @delandtj . he is travelling this week and will not be able to send the mac addresses before friday.

**Comment by andhartl** [comment]
@delandtj please send the mac addresses to @ramezsaeed as Maged is leaving the company. thx
